MorphicServer/Endpoint.cs

```diff
@@ -340,6 +348,122 @@ public async Task WithTransaction(Func<Database.Session, Task> operations)
                 throw new HttpError(HttpStatusCode.InternalServerError);
             }
         }
+
+        public static string? ClientIpFromRequest(HttpRequest request)
```

Comment:
This is the kind of thing I'd make into an `HttpRequest` extension.  Personal preference, helps de-clutter `Endpoint`

Reply:
I have mixed feelings about injecting methods into another class! Seems... weird. But I'm game if you show me how.